lambdaclass/mirra_backend
# [GH-549] Fix valtimer teleporting inside polygons

## Motivation

Whenever you used valtimers teleport ability between two polygon obstacles you could get inside of a collider in the map

In a nutshell this error was due to how we process collision resolution with polygon and the fact that colliders in the map are composed of several polygons due to the fact that we can't have concave polygons. 
Since the collition resolution algorithm always return the closest vector that would solve the collition this could result in a movement that gets you inside another polygon, resulting in ending up inside the obstacles.

The fix will work by filtering movements that would result in ending up inside another obstacle, something like:

![image](https://github.com/lambdaclass/mirra_backend/assets/106101218/668d0621-3bcb-47f3-8d18-9f8638ff52ee)


This was the bug present on main:

https://github.com/lambdaclass/mirra_backend/assets/67444519/d97bfeaa-5d2c-42cc-9e2a-ec8ec70d5a37



Closes #549 

08/05 update:
There was another case were an axis was invalid if the result would make the player to go outside the map 

## Summary of changes

- Added line to polygon collision detection
- Filter invalid axis when processing collision resolution with sat
- Fixed cannon stones polygon vertex to remove hole in the middle

## How to test it?
Start a game with valtimer and try to use the teleport inside a rock
Also try to walk to obstacles and check that it feels correct

## Checklist
